src/overlays/actors/ovl_Obj_Boyo/z_obj_boyo.c

```diff
@@ -19,22 +26,183 @@ const ActorInit Obj_Boyo_InitVars = {
+    this->animatedMaterial = Lib_SegmentedToVirtual(&object_boyo_Matanimheader_000E88);
+}
+
+void ObjBoyo_Destroy(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    ObjBoyo* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void ObjBoyo_PushPlayer(ObjBoyo* this, Actor* actor) {
+    ((Player*)actor)->pushedSpeed = 30.0f;
+    ((Player*)actor)->pushedYaw = (s16)this->actor.yawTowardsPlayer;
+}
+
+void ObjBoyo_PushPirate(ObjBoyo* this, Actor* actor) {
+    EnKaizoku* kaizoku = (EnKaizoku*)actor;
+
+    kaizoku->unk_2F0 = 30.0f;
+    kaizoku->unk_2F4 = Actor_WorldYawTowardActor(&this->actor, &kaizoku->picto.actor);
+}
+
+void ObjBoyo_ExplodeBomb(ObjBoyo* this, Actor* actor) {
+    EnBom* bom = (EnBom*)actor;
+
+    bom->timer = 0;
+}
+
+/*
+ * Find the Actor we collided with.
+ */
+Actor* ObjBoyo_FindCollidedActor(ObjBoyo* this, PlayState* play, s32* index) {
+    Actor* collidedActor;
+    ObjBoyoUnkStruct* data;
+    s32 counter;
+
+    if (this->collider.base.ocFlags2 & OC2_HIT_PLAYER) {
+        *index = 0;
+        // Returns first actor of the actor List for actors of type Player.
+        // In short it gets the player's Actor*.
+        return (s32)play->actorCtx.actorLists[2].first;
+    }
+    data = gCollisionHandling + 1;
+    if (this->collider.base.ocFlags1 & OC1_HIT) {
+        collidedActor = this->collider.base.oc;
+        for (counter = 1; counter != 3; ++counter) {
+            if (collidedActor->id == (*data).id) {
+                *index = counter;
+                return collidedActor;
+            }
+            data += 1;
+        }
+    }
+    return 0;
+}
+
+void ObjBoyo_Update(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    ObjBoyo* this = THIS;
+    Actor* collidedActor;
+    s32 index;
+
+    collidedActor = ObjBoyo_FindCollidedActor(this, play, &index);
+
+    if (collidedActor != 0) {
+        gCollisionHandling[index].colHandler(this, collidedActor);
+        // TODO: find out what all of these are.
+
+        this->unk194 = 0x64;
+        this->unk196 = 3;
+        this->unk198 = 0.01f;
+        this->unk19C = 0.03f;
+        this->unk1A4 = 0x3F40;
+        this->unk1A6 = 0x7D0;
+        this->unk1A8 = 0;
+        this->unk1A0 = 0.03f;
+        this->unk1AA = 0x2DF7;
+        this->unk1AC = 0x258;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boyo/ObjBoyo_Destroy.s")
+    if (this->unk194 > 0) { /* compute new scaling */
+        // this computation might help finding what those values are.
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boyo/func_809A5DC0.s")
+        this->unk194 -= this->unk196;
+        this->unk1AA += this->unk1AC;
+        this->unk1A8 += this->unk1AA;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boyo/func_809A5DE0.s")
+        this->actor.scale.x = this->actor.scale.z =
+            (Math_CosS((s16)(this->unk1A8 + this->unk1A4)) * (f32)this->unk194 * this->unk19C * this->unk198) + 0.1f;
```

Comment:
```suggestion
            (Math_CosS(this->unk1A8 + this->unk1A4) * (f32)this->unk194 * this->unk19C * this->unk198) + 0.1f;
```
Shouldn't need the `s16` cast here. Could you also change it for the one just below.